haystack/components/preprocessors/csv_document_splitter.py

```diff
@@ -97,7 +112,11 @@ def run(self, documents: List[Document]) -> Dict[str, List[Document]]:
                 split_documents.append(document)
                 continue
 
-            if self.row_split_threshold is not None and self.column_split_threshold is None:
+            if self.split_mode == "row-wise":
+                # each row is a separate sub-table
+                split_dfs = self._split_by_row(df=df)
+
+            elif self.row_split_threshold is not None and self.column_split_threshold is None:
```

Comment:
Since we have split_mode now, could we put all threshold related stuff under it's own if statement, so something like 
```python
if self.split_mode == "row-wise":
...
elif self.split_mode == "threshold":
...
else:
    raise ValueError(...)
```